x/superfluid/keeper/stake.go

```diff
@@ -61,62 +60,55 @@ func (k Keeper) RefreshIntermediaryDelegationAmounts(ctx sdk.Context) {
 			continue
 		}
 
-		// undelegate full amount from the validator
 		delegation, found := k.sk.GetDelegation(ctx, mAddr, valAddress)
+		if !found {
+			k.Logger(ctx).Error(fmt.Sprintf("delegation not found for %s with %s", mAddr.String(), acc.ValAddr))
+			continue
+		}
+
+		currentAmount := validator.TokensFromShares(delegation.Shares).RoundInt()
 
-		if found {
-			returnAmount, err := k.sk.Unbond(ctx, mAddr, valAddress, delegation.Shares)
+		refreshedAmount := k.GetExpectedDelegationAmount(ctx, acc)
+
+		if refreshedAmount.GT(currentAmount) {
+			//need to mint and delegate
+			adjustment := refreshedAmount.Sub(currentAmount)
+			coins := sdk.NewCoins(sdk.NewCoin(bondDenom, adjustment))
+			err = k.bk.MintCoins(ctx, minttypes.ModuleName, coins)
```

Comment:
What is the module we are minting to? Why not mint to the intermediate account or a single superfluid temporary 'holding' account?

Reply:
Not that big of a deal, lets just file into an issue as a minor thing to handle.

Reply:
I've been switching these to use the superfluid module instead of mint module, missed this one but just fixed it.